scripts/configure_valhalla.sh

```diff
@@ -24,7 +24,7 @@ fi
 do_elevation="False"
 do_admins="False"
 do_timezones="False"
-if [[ "${build_elevation}" == "True" ]] || [[ "${build_elevation}" == "Force" ]] || ! ([[ -z "${min_x}" ]] && [[ -z "${min_y}" ]] && [[ -z "${max_x}" ]] && [[ -z "${max_y}" ]]); then
+if ([[ "${build_elevation}" == "True" ]] || [[ "${build_elevation}" == "Force" ]]) && ! ([[ -z "${min_x}" ]] || [[ -z "${min_y}" ]] || [[ -z "${max_x}" ]] || [[ -z "${max_y}" ]]); then
```

Comment:
All right. I thought the variables max_x, min_x, max_y and min_y are just to control the bbox controlling the zone we want to have elevation data for but it seems it is also there to remain backward compatibility.
In your latest suggestion if we set build_elevation=False but one of the aforementioned variables are set, the do_elevation will still become zero and the elevation data will be downloaded neglecting what we had set before.

Reply:
I think we need to make a distinction between height service and routing like services (the user should be able to turn off elevation for routing and yet have it available separately for height.)

Reply:
> if we set build_elevation=False but one of the aforementioned variables are set, the do_elevation will still become zero

That's right. Probably that was the reason why it is the way it is and I was too quick with the suggestion, so maybe we shouldn't touch this at all?

> the user should be able to turn off elevation for routing and yet have it available separately for height.

Hm, I haven't met anyone who's using Valhalla for only height (also I don't think anyone would arrive at Valhalla with only that in mind). And if it's also routing, I don't see why anyone would want to build a graph without elevation information, when the data is already there for the height service.

I could be convinced to include such logic into the repo if someone's using the image that way and wants to PR.

Reply:
No, what I meant is that I think we need to separate the bbox variables for height and for routing. Because the use might want to use the routing just within a specific region while he might use a different region (whole world) just in height.

Also if we leave the code intact the problem still exists. If we set all bbox variables and set build_elevation to False, because the third part of the if condition (related to bbox variables) return True, the do_elevation will be set to True which is again the opposite thing we wanted. I made these changes so that of someone don't want to build elevation tiles (by setting build_elevation = False it wont be built.

Reply:
Sure, generally you’re right. As of right now (state of the PR), my main point is: those coords are not used (and more importantly not documented) anywhere, so it’s, for me at least, _only_ for backwards compatibility.

The use case you’re describing is very niche, but like I said, I could be convinced with a PR, if that doesn’t it make much more complex. We could use those coords for what you’re describing (it’s also long enough that we deprecated them to use them in another way). If you want to update the PR with that, I’ll review for sure.